### PR TITLE
added npm compatibility

### PR DESCRIPTION
**Pull request by itsjavi**
- Cleans up / complements some dot files
- Package name and lib file consistency: in lower case.
- Adds compatiblity with [npm publishing](https://www.npmjs.com/package/melonjs)
   - Usual steps to publish:
     - Increment `version` in `package.json` & commit and push the change
     - run `grunt` and then `npm publish`

### CODE REVIEW

**Comment by obiot** [comment]
awesome, thanks for that ! One improvement that I see is certainly to include that call to `nom publish` in our grunt release task (@parasyte)

File: .editorconfig [hunk 1]
```diff
@@ -19,3 +19,7 @@ indent_size = 2
 [Gruntfile.js]
 indent_style = space
 indent_size = 2
+
+[Gruntfile.js]
+indent_style = space
+indent_size = 2
```
Comment (obiot) [inline .editorconfig]: this is duplicated now :)

Comment (itsjavi) [reply .editorconfig]: true, I fixed that :)

**Comment by itsjavi** [comment]
I added @parasyte as collaborator in the npm package. I could not find you there @obiot

**Pull request merged**